diff --git a/meson.build b/meson.build
index e3839cf5..279d655f 100644
--- a/meson.build
+++ b/meson.build
@@ -266,103 +266,12 @@ elif is_dragonfly or is_freebsd or is_netbsd or is_openbsd
     endif
 endif
 
-if true
-    add_project_arguments('-DHAVE_SWAY', language: 'cpp')
-    src_files += files(
-        'src/modules/sway/ipc/client.cpp',
-        'src/modules/sway/bar.cpp',
-        'src/modules/sway/mode.cpp',
-        'src/modules/sway/language.cpp',
-        'src/modules/sway/window.cpp',
-        'src/modules/sway/workspaces.cpp',
-        'src/modules/sway/scratchpad.cpp'
-    )
-    man_files += files(
-        'man/waybar-sway-language.5.scd',
-        'man/waybar-sway-mode.5.scd',
-        'man/waybar-sway-scratchpad.5.scd',
-        'man/waybar-sway-window.5.scd',
-        'man/waybar-sway-workspaces.5.scd',
-    )
-endif
-
 if true
     add_project_arguments('-DHAVE_WLR_TASKBAR', language: 'cpp')
     src_files += files('src/modules/wlr/taskbar.cpp')
     man_files += files('man/waybar-wlr-taskbar.5.scd')
 endif
 
-if wayland_protos.version().version_compare('>=1.39')
-    add_project_arguments('-DHAVE_EXT_WORKSPACES', language: 'cpp')
-    src_files += files(
-        'src/modules/ext/workspace_manager.cpp',
-        'src/modules/ext/workspace_manager_binding.cpp',
-    )
-    man_files += files(
-        'man/waybar-ext-workspaces.5.scd',
-    )
-endif
-
-if true
-    add_project_arguments('-DHAVE_RIVER', language: 'cpp')
-    src_files += files(
-        'src/modules/river/layout.cpp',
-        'src/modules/river/mode.cpp',
-        'src/modules/river/tags.cpp',
-        'src/modules/river/window.cpp',
-    )
-    man_files += files(
-        'man/waybar-river-layout.5.scd',
-        'man/waybar-river-mode.5.scd',
-        'man/waybar-river-tags.5.scd',
-        'man/waybar-river-window.5.scd',
-    )
-endif
-
-if true
-    add_project_arguments('-DHAVE_DWL', language: 'cpp')
-    src_files += files('src/modules/dwl/tags.cpp')
-    src_files += files('src/modules/dwl/window.cpp')
-    man_files += files('man/waybar-dwl-tags.5.scd')
-    man_files += files('man/waybar-dwl-window.5.scd')
-endif
-
-if true
-    add_project_arguments('-DHAVE_HYPRLAND', language: 'cpp')
-    src_files += files(
-        'src/modules/hyprland/backend.cpp',
-        'src/modules/hyprland/language.cpp',
-        'src/modules/hyprland/submap.cpp',
-        'src/modules/hyprland/window.cpp',
-        'src/modules/hyprland/windowcount.cpp',
-        'src/modules/hyprland/workspace.cpp',
-        'src/modules/hyprland/workspaces.cpp',
-        'src/modules/hyprland/windowcreationpayload.cpp',
-    )
-    man_files += files(
-        'man/waybar-hyprland-language.5.scd',
-        'man/waybar-hyprland-submap.5.scd',
-        'man/waybar-hyprland-window.5.scd',
-        'man/waybar-hyprland-workspaces.5.scd',
-    )
-endif
-
-if get_option('niri')
-    add_project_arguments('-DHAVE_NIRI', language: 'cpp')
-    src_files += files(
-        'src/modules/niri/backend.cpp',
-        'src/modules/niri/language.cpp',
-        'src/modules/niri/window.cpp',
-        'src/modules/niri/workspaces.cpp',
-        'src/modules/niri/workspace.cpp',
-    )
-    man_files += files(
-        'man/waybar-niri-language.5.scd',
-        'man/waybar-niri-window.5.scd',
-        'man/waybar-niri-workspaces.5.scd',
-    )
-endif
-
 if get_option('mango')
     add_project_arguments('-DHAVE_MANGO', language: 'cpp')
     src_files += files(
@@ -382,18 +291,6 @@ if get_option('mango')
     )
 endif
 
-if true
-    add_project_arguments('-DHAVE_WAYFIRE', language: 'cpp')
-    src_files += files(
-        'src/modules/wayfire/backend.cpp',
-        'src/modules/wayfire/window.cpp',
-        'src/modules/wayfire/workspaces.cpp',
-    )
-    man_files += files(
-        'man/waybar-wayfire-window.5.scd',
-        'man/waybar-wayfire-workspaces.5.scd',
-    )
-endif
 
 if get_option('login-proxy')
     add_project_arguments('-DHAVE_LOGIN_PROXY', language: 'cpp')
